### ISSUE TITLE
[Android] Exception java.lang.ClassCastException: java.util.HashMap cannot be cast to com.facebook.react.bridge.ReadableMap

### ISSUE THREAD

**Issue by ghost** [DataDog/dd-sdk-reactnative#651]
### Stack trace

```
Exception java.lang.ClassCastException: java.util.HashMap cannot be cast to com.facebook.react.bridge.ReadableMap
  at com.datadog.reactnative.DdSdkConfigurationExtKt.asFirstPartyHosts (DdSdkConfigurationExt.kt:279)
  at com.datadog.reactnative.DdSdkConfigurationExtKt.asDdSdkConfiguration (DdSdkConfigurationExt.kt:46)
  at com.datadog.reactnative.DdSdkImplementation.initialize (DdSdkImplementation.kt:38)
  at com.datadog.reactnative.DdSdk.initialize (DdSdk.kt:31)
  at java.lang.reflect.Method.invoke
  at com.facebook.react.bridge.JavaMethodWrapper.invoke (JavaMethodWrapper.java:372)
  at com.facebook.react.bridge.JavaModuleWrapper.invoke (JavaModuleWrapper.java:188)
  at com.facebook.jni.NativeRunnable.run
  at android.os.Handler.handleCallback (Handler.java:938)
  at android.os.Handler.dispatchMessage (Handler.java:99)
  at com.facebook.react.bridge.queue.MessageQueueThreadHandler.dispatchMessage (MessageQueueThreadHandler.java:27)
  at android.os.Looper.loopOnce (Looper.java:201)
  at android.os.Looper.loop (Looper.java:288)
  at com.facebook.react.bridge.queue.MessageQueueThreadImpl$4.run (MessageQueueThreadImpl.java:228)
  at java.lang.Thread.run (Thread.java:920)
```

### Volume

100%

### Affected SDK versions

2.3.1

### Latest working SDK version

2.2.0

### Does the crash manifest in the latest SDK version?

Yes

### React Native Version

0.71.16

### Device Information

Android only

Google play store pre-launch report saw crashes on every test device
<img width="400" alt="Screenshot 2024-04-10 at 7 15 32 AM" src="https://github.com/DataDog/dd-sdk-reactnative/assets/64432374/8e038e53-e6fa-4680-9662-8a59751040d0">

**Comment by jacobhummel** [comment]
Same issue on Expo 50 upgrade on Android

"react-native": "0.73.6",
"expo": "^50.0.15",
"@datadog/mobile-react-native": "^2.3.1",
"@datadog/mobile-react-navigation": "^2.3.1",
"expo-datadog": "^50.2.0",

**Comment by krutarth-h-dave** [comment]
I am also facing the same issue as OP.

**Comment by jacobhummel** [comment]
Resolved for now by downgrading

"@datadog/mobile-react-native": "2.2.0",
"@datadog/mobile-react-navigation": "2.2.0",

**Comment by krutarth-h-dave** [comment]
> Resolved for now by downgrading
> 
> "@datadog/mobile-react-native": "2.2.0", "@datadog/mobile-react-navigation": "2.2.0",

+1.

**Comment by marco-saia-datadog** [comment]
Hello! The issue has been introduced with 2.3.0 and occurs when setting `firstPartyHosts` in your configuration file. The fix is ready and will soon be released. Here is the PR: https://github.com/DataDog/dd-sdk-reactnative/pull/644

**Comment by marco-saia-datadog** [comment]
@bhandanyan-nomad @jacobhummel @krutarth-h-dave 

I am happy to inform you that the fix has been released in 2.3.2, and you can now safely update to the new version!

### v2.3.2
* **FIX**: Android Runtime Error when using `firstPartyHosts` initialization setting
* **FIX**: `JSRefreshRateMonitor` iOS memory leak
